template-processors/base/bin/resourceManager.sh

```diff
@@ -14,59 +14,54 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -o nounset
-set -o errexit
+set -euxo pipefail
 
-# this is needed becasue we want the current namespace to be set as default if a namespace is not specified.
+# this is needed because we want the current namespace to be set as default if a namespace is not specified.
 function setContext {
-  $kubectl config set-context current --namespace=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+  $kubectl config set-context current --namespace="$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)"
   $kubectl config use-context current
 }
 
 function kube {
-  $kubectl -s https://kubernetes.default.svc:443  --token $(cat /var/run/secrets/kubernetes.io/serviceaccount/token) --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt $@
+  $kubectl \
+    -s https://kubernetes.default.svc:443 \
+    --token "$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+    "$@"
 }
 
 function deleteResources {
-    #first we need to delete the GitOpsConfig resources whose finalizer might not work otherwise
-    for file in $(find $MANIFEST_DIR -iregex '.*\.yaml'); do
-      cat $file | yq 'select(.kind == "GitOpsConfig")' | kube delete -f - --wait=true
-    done
-    set +u
-    kube delete -R -f $MANIFEST_DIR
-    set -u
+  #first we need to delete the GitOpsConfig resources whose finalizer might not get called otherwise
+  for file in $(find "$MANIFEST_DIR" -iregex '.*\.yaml'); do
+    cat "$file" | yq 'select(.kind == "GitOpsConfig")' | kube delete -f - --wait=true
+  done
+  kube delete -R -f "$MANIFEST_DIR"
 }
 
 function createUpdateResources {
-  if [ $CREATE_MODE == "CreateOrMerge" ]; then
-    kube apply -R -f $MANIFEST_DIR
-  fi
-  if [ $CREATE_MODE == "CreateOrUpdate" ]; then
-    set +u
-    kube create -R -f $MANIFEST_DIR
-    set -u
-    kube update -R -f $MANIFEST_DIR
-  fi
-  if [ $CREATE_MODE == "Patch" ]; then
-    kube patch -R -f $MANIFEST_DIR
-  fi
-
+  case "$CREATE_MODE" in
+    CreateOrMerge)
+      kube apply -R -f "$MANIFEST_DIR"
+      ;;
+    CreateOrUpdate)
+      kube create -R -f "$MANIFEST_DIR"
+      kube update -R -f "$MANIFEST_DIR"
```

Comment:
What about update operation? Did it clarify?

Reply:
I'm tracking this in a separate PR #149 so that any one of them does not block the other.

Reply:
cool